src/lib/swapper/swappers/ThorchainSwapper/utils/getSignTxFromQuote.ts

```diff
@@ -0,0 +1,136 @@
+import type { ChainId } from '@shapeshiftoss/caip'
+import { CHAIN_NAMESPACE, fromAssetId } from '@shapeshiftoss/caip'
+import type {
+  CosmosSdkBaseAdapter,
+  EvmChainAdapter,
+  utxo,
+  UtxoBaseAdapter,
+} from '@shapeshiftoss/chain-adapters'
+import { getChainAdapterManager } from 'context/PluginProvider/chainAdapterSingleton'
+import type { TradeQuote, UnsignedTx } from 'lib/swapper/api'
+import { getCosmosTxData } from 'lib/swapper/swappers/ThorchainSwapper/cosmossdk/getCosmosTxData'
+import type { ThorEvmTradeQuote } from 'lib/swapper/swappers/ThorchainSwapper/getThorTradeQuote/getTradeQuote'
+import type {
+  ThorCosmosSdkSupportedChainId,
+  ThorUtxoSupportedChainId,
+} from 'lib/swapper/swappers/ThorchainSwapper/ThorchainSwapper'
+import { getThorTxInfo } from 'lib/swapper/swappers/ThorchainSwapper/utxo/utils/getThorTxData'
+import { DEFAULT_SLIPPAGE } from 'lib/swapper/swappers/utils/constants'
+import { assertUnreachable } from 'lib/utils'
+import { createSignTxInput } from 'lib/utils/evm'
+
+import { isNativeEvmAsset } from '../../utils/helpers/helpers'
+
+export const getSignTxFromQuote = async ({
+  tradeQuote: quote,
+  receiveAddress,
+  affiliateBps = '0',
+  chainSpecific,
+  buyAssetUsdRate,
+  feeAssetUsdRate,
+  from,
+  xpub,
+  supportsEIP1559,
+}: {
+  tradeQuote: TradeQuote<ChainId>
+  receiveAddress: string
+  affiliateBps?: string
+  chainSpecific?: utxo.BuildTxInput
+  buyAssetUsdRate: string
+  feeAssetUsdRate: string
+  supportsEIP1559: boolean
+} & ({ from: string; xpub?: never } | { from?: never; xpub: string })): Promise<UnsignedTx> => {
+  const { recommendedSlippage: slippageTolerance = DEFAULT_SLIPPAGE } = quote
```

Comment:
this is using a default slippage shared across all swappers - is this always in the same units or does this need to be specific per swapper?

Reply:
can we use this `getDefaultSlippagePercentageForSwapper`?

Reply:
This util is for THORChain so we are good here:

https://github.com/shapeshift/web/blob/1b8b2449619c00108c6857c2baed70e4d0b3e16d/src/constants/constants.ts#L11-L19

With that being said, might as well use the helper for consistency if/when we implement this for other swappers - b6c45e02bff070dac2c6922afb4143da988ea18f